shared/utils/logging_wrappers.hpp

```diff
@@ -0,0 +1,82 @@
+#pragma once
+
+#include "logging.hpp"
+
+#include "base-wrapper-type.hpp"
+#include "typedefs-wrappers.hpp"
+#include "typedefs-string.hpp"
+#include "typedefs-array.hpp"
+#include "typedefs-list.hpp"
+
+template<typename T>
+struct fmt::formatter<SafePtr<T>> {
+
+// Formats the point p using the parsed format specification (presentation)
+// stored in this formatter.
+template<typename FormatContext>
+auto format(SafePtr<T> const& p, FormatContext &ctx) -> decltype(ctx.out()) {
+// ctx.out() is an output iterator to write to.
+return format_to(ctx.out(), "{}|{}", p.operator bool(), fmt::ptr(p));
+}
+};
+
+template<typename T>
+struct fmt::formatter<SafePtrUnity<T>> {
+
+// Formats the point p using the parsed format specification (presentation)
+// stored in this formatter.
+template<typename FormatContext>
+auto format(SafePtr<T> const& p, FormatContext &ctx) -> decltype(ctx.out()) {
```

Comment:
Should be `SafePtrUnity<T>`, also see comments on SafePtr's formatter. Ex, what do we do with const? What about the internal handle/etc.

Except in this case, checking the internal handle is not enough, we should also check the unity validity of this pointer. Doing this would be nice if we can do so in a fast way (ex, without any unity icalls) and spitting out three different messages would be nice:
1. no handle
2. not unity valid
3. valid pointer, here is the address